### ISSUE TITLE
archetypes mode with wrong input dir = non-obvious crash

### ISSUE THREAD

**Issue by sbailey** [desihub/redrock#243]
When rrdesi is provided an incorrect archetypes dir (doesn't exist or otherwise unreadable), redrock silently does the wrong thing and proceeds until it hits a non-obvious crash.  It would be less confusing to crash immediately if no archetypes are found (or readable) within the specified archetypes directory.
```
rrdesi --archetypes /does/not/exist \
    -i $CFS/desi/spectro/redux/fuji/tiles/cumulative/80606/20201219/coadd-0-80606-thru20201219.fits \
    --targetids 39627646572697743,39627652591519496 -o test.fits -d test.hd5
...
Traceback (most recent call last):
  File "/global/common/software/desi/users/sjbailey/redrock/bin/rrdesi", line 10, in <module>
    desi.rrdesi(comm=None)
  File "/global/common/software/desi/users/sjbailey/redrock/py/redrock/external/desi.py", line 886, in rrdesi
    raise err
  File "/global/common/software/desi/users/sjbailey/redrock/py/redrock/external/desi.py", line 805, in rrdesi
    scandata, zfit = zfind(targets, dtemplates, mpprocs,
  File "/global/common/software/desi/users/sjbailey/redrock/py/redrock/zfind.py", line 414, in zfind
    spectype = [ el[0].split(':::')[0] for el in tmp['fulltype'] ] #el is a list with one element (corresponding to each minima)
KeyError: 'fulltype'
```

**Comment by abhi0395** [comment]
Hi @sbailey 

I guess the reason this happens is that redrock looks for archetype files in the very last steps when it actually looks for the best archetypes for best-fit redshifts. Until then it doesn't actually look if the file exists.

I think this can be solved by checking the `args.archetypes` in the very beginning. If the archetypes are provided and it does not exist the rrdesi should just exit, otherwise continue.

I just created a pull request with the branch name `missing_archetypes` that solves this issue.

I re-ran the same script on `missing_archetypes` and it succeeds if the provided file exists and raise IOError if the file does not exist and exits.

`rrdesi --archetypes /does/not/exist  -i $CFS/desi/spectro/redux/fuji/tiles/cumulative/80606/20201219/coadd-0-80606-thru20201219.fits --targetids 39627646572697743,39627652591519496 -o test.fits -d test.hd5`

gives

`ERROR: can't find archetypes_dir`

But if the file exists: 

`rrdesi --archetypes/global/homes/a/abhijeet/software/desisoft/redrock-archetypes -i $CFS/desi/spectro/redux/fuji/tiles/cumulative/80606/20201219/coadd-0-80606-thru20201219.fits --targetids 39627646572697743,39627652591519496 -o test.fits -d test.hd5`

it succeeds!

While the `main` branch fails if run:

`rrdesi --archetypes /does/not/exist -i $CFS/desi/spectro/redux/fuji/tiles/cumulative/80606/20201219/coadd-0-80606-thru20201219.fits --targetids 39627646572697743,39627652591519496 -o test.fits -d test.hd5`

gives error:

```
--- Process 0 raised an exception ---
Proc 0: Traceback (most recent call last):
Proc 0:   File "/global/homes/a/abhijeet/software/desisoft/redrock/py/redrock/external/desi.py", line 805, in rrdesi
    scandata, zfit = zfind(targets, dtemplates, mpprocs,
Proc 0:   File "/global/homes/a/abhijeet/software/desisoft/redrock/py/redrock/zfind.py", line 414, in zfind
    spectype = [ el[0].split(':::')[0] for el in tmp['fulltype'] ] #el is a list with one element (corresponding to each minima)
Proc 0: KeyError: 'fulltype'

Traceback (most recent call last):
  File "/global/homes/a/abhijeet/software/desisoft/redrock/bin/rrdesi", line 10, in <module>
    desi.rrdesi(comm=None)
  File "/global/homes/a/abhijeet/software/desisoft/redrock/py/redrock/external/desi.py", line 886, in rrdesi
    raise err
  File "/global/homes/a/abhijeet/software/desisoft/redrock/py/redrock/external/desi.py", line 805, in rrdesi
    scandata, zfit = zfind(targets, dtemplates, mpprocs,
  File "/global/homes/a/abhijeet/software/desisoft/redrock/py/redrock/zfind.py", line 414, in zfind
    spectype = [ el[0].split(':::')[0] for el in tmp['fulltype'] ] #el is a list with one element (corresponding to each minima)
KeyError: 'fulltype'


```

**Comment by moustakas** [comment]
Looks like this was fixed in #244.
